tests/agent/test_agent.py

```diff
@@ -40,9 +41,70 @@ def finalize(job_id):
 
     api.set_job_transition(job_id, ExecutorState.FINALIZED, hook=finalize)
     assert job_id not in api.tracker["finalize"]
-    main.handle_run_job_task(task, api)
+    main.handle_single_task(task, api)
     assert job_id in api.tracker["finalize"]
     task = controller_task_api.get_task(task.id)
     assert task.agent_stage == ExecutorState.FINALIZED.value
     assert task.agent_complete
     assert "results" in task.agent_results
+
+    spans = get_trace("agent_loop")
+    # one span each time we called main.handle_single_task
+    assert len(spans) == 3
+
+    common_attribute_keys = {
+        "backend",
+        "task_type",
+        "task",
+        "task_created_at",
+        "initial_job_status",
+        "job",
+        "job_request",
+        "workspace",
+        "repo_url",
+        "commit",
+        "action",
+        "job_created_at",
+        "image",
+        "args",
+        "inputs",
+        "allow_database_access",
+        "cpu_count",
+        "memory_limit",
+        "final_job_status",
+        "complete",
+    }
+    for span in spans[0:1]:
+        assert span.name == "LOOP_TASK"
+        # check we have the keys we expect (the final span also contains results
+        # so has additional attribute keys)
+        assert set(span.attributes.keys()) == common_attribute_keys
+
+        # attributes added from the task
+        assert span.attributes["backend"] == "test"
+        assert span.attributes["task_type"] == "RUNJOB"
+        assert span.attributes["task"] == task.id
+        assert span.attributes["task_created_at"] == task.created_at * 1e9
+        # attributes added from the job
+        assert span.attributes["job"] == job_id
+
+    assert set(spans[2].attributes.keys()) == common_attribute_keys.union(
+        {
+            # results included on the final span
+            "unmatched_patterns",
+            "image_id",
+            "unmatched_outputs",
+            "message",
+            "exit_code",
+        }
+    )
+
+    assert spans[0].attributes["initial_job_status"] == "UNKNOWN"
```

Comment:
What do you think about this test just testing the correct statuses, and not all the attributes, and move the testing of attributes into a test_tracing.py module?

Reply:
The various tracing assertions in the controller tests I was working on also felt a bit weird, but it didn't seem the right time to be worrying about them and so I left them as they were all passing.